damnit/gui/zulip_messenger.py

```diff
@@ -0,0 +1,304 @@
+        
+        self.cancel_button.clicked.connect(self.reject)
+        self.ok_button.clicked.connect(self.handle_form)
+
+
+        
+    def check_for_changes(self, include_streams = True):
+        changes = self.messenger.email != self.edit_email.text() or \
+            self.messenger.key != self.edit_key.text() or\
+            include_streams*(
+                self.messenger.stream != self._edit_stream.text() or 
+                self.messenger.topic != self._edit_topic.text())
+                
+        if changes:                
+            self.messenger.email, self.messenger.key, self.messenger.stream, self.messenger.topic = \
+            self.edit_email.text(), self.edit_key.text(), self._edit_stream.text(), self._edit_topic.text()
+            self.save_config_file()
+            
+        return changes
+    
+    def handle_form(self):
+        if self.check_for_changes(): 
+            self.update_client()
+        self._send_msg()
+        
+    def update_client(self):
+        try:
+            self.messenger.client = zulip.Client(config_file=self.config_path)
+        except Exception as exc:
+            self.show_msg(traceback.format_exc())
+            log.error(exc, exc_info=True)
+        
+    def show_msg(self, msg, level = 'error'):
+        if level == 'error':
+            self.output.setStyleSheet("""QPlainTextEdit { color: red };""")
+        elif level == 'warning':
+            self.output.setStyleSheet("""QPlainTextEdit { color: orange };""")
+        elif level == 'debug':
+            self.output.setStyleSheet("""QPlainTextEdit { color: gray };""")
+
+        if msg == 'Invalid API key':
+            msg = msg + ' ' + 'or email address'
+
+        self.output.setPlainText(msg)
+        
+    def show_default_msg(self):
+        self.show_msg('Logs will be printed here', level='debug')
+    
+    def _send_msg(self):     
+        if self.kind == 'table':
+            if self.edit_title.text() != '':
+                self.msg = f"### {self.edit_title.text()}" + "\n" + self.msg
+        
+        elif self.kind == 'figure':
+            try:
+                upload = self.messenger.client.upload_file(self.msg)
+                if upload['result'] == 'error':
+                    self.show_msg(upload['msg'])
+                    return
+                
+            except Exception as exc:
+                self.show_msg(traceback.format_exc())
+                log.error(exc, exc_info=True)
+                return 
+                
+            self.msg = f"[{self.edit_title.text()}]({upload['uri']})"
+            
+        request =  {
+        "type": "stream",
+        "to": f"{self.messenger.stream}",
+        "topic": f"{self.messenger.topic}",
+        "content": f"{self.msg}"
+        }
+        
+        try:
+            response = self.messenger.client.send_message(request)
+        except Exception as exc:
+            response = {'result' : '', 'msg': f"{exc}"}
```

Comment:
Looks like this is a honeypot of connection errors. The traces seem to be large and unfortunately I cannot see it further on the logs text box as it is disabled. You could either catch the known exceptions and pass a concise message or enable the text box (and set it as read-only) such that one can scroll through the error message.